scripts/get-ldflags.sh

```diff
@@ -8,19 +13,21 @@ if [ "$MODE" != "LDFLAG" ] && [ "$MODE" != "ECHO" ]; then
 fi
 
 # $(git rev-parse --short HEAD) \"$(git log -1 --format=%cI-%h)\"
-export GIT_REVISION=$(git log -1 --format=%cI-%h) # use commit date time and hash: e.g. 2021-07-27 19:36:53 -0700-40c5f1896, doc: https://git-scm.com/docs/pretty-formats
-export GIT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
-export BUILD_DATE=$(date '+%F-%T') # outputs something in this format 2017-08-21-18:58:45
-export BUILD_TS_UNIX=$(date '+%s') # second since epoch
-export BASE_PACKAGE=github.com/uber/cadence/common/metrics
+GIT_REVISION=$(git log -1 --format=%cI-%h) # use commit date time and hash: e.g. 2021-07-27 19:36:53 -0700-40c5f1896, doc: https://git-scm.com/docs/pretty-formats
+GIT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
+BUILD_DATE=$(date '+%F-%T') # outputs something in this format 2017-08-21-18:58:45
+BUILD_TS_UNIX=$(date '+%s') # second since epoch
+BASE_PACKAGE=github.com/uber/cadence/common/metrics
 if [ -z ${CADENCE_RELEASE_VERSION} ]; then
   # If not set CADENCE_RELEASE_VERSION, then use the most recent tag.
-  export RELEASE_VERSION=$(git describe --tags --abbrev=0 --dirty 2>/dev/null || echo unknown)
+  RELEASE_VERSION=$(git describe --tags --abbrev=0 --dirty 2>/dev/null || echo unknown)
 else
   # If passing a CADENCE_RELEASE_VERSION explicitly, then use it
-  export RELEASE_VERSION=${CADENCE_RELEASE_VERSION}
+  RELEASE_VERSION=${CADENCE_RELEASE_VERSION}
 fi
 
+# var-setting and exporting must be done separately or `set -e` does not catch errors
+export GIT_REVISION GIT_BRANCH BUILD_DATE BUILD_TS_UNIX BASE_PACKAGE RELEASE_VERSION
```

Comment:
TIL.  shellcheck catches this, I tried to add a stable version to our pipeline, but failed several times and don't feel like spending more hours figuring out the internals of buildkite :|

it'd be pretty easy to add to our makefile, but getting a stable version across local and ci is probably hard.  also probably worthwhile either way, but still.

Reply:
why do we need these exported?

Reply:
since it's just echoing: good point!  we do not.
I can just drop that in here.